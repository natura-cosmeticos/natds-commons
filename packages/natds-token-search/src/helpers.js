import {
  pipe, unless, objOf, zipObj, mapObjIndexed, isNil,
  pickBy, or, contains, toLower, toString
} from 'ramda'

export const buildTokensForBrand = (brand) => pipe(
  zipObj(['light', 'dark']),
  objOf(brand)
)

const setAttribute = (element) => (attribute, key) => element.setAttribute(key, attribute)
const setContent = (element) => (content) => {
  element.innerHTML = content; // eslint-disable-line
}

export const createElement = (element, attributes, content, className) => {
  const el = document.createElement(element)

  unless(isNil, mapObjIndexed(setAttribute(el)))(attributes)
  unless(isNil, setContent(el))(content)

  if (className) {
    el.classList.add(className)
  }

  return el
}

const compareToSearch = (search) => pipe(toString, toLower, contains(toLower(search)))

const searchRules = (search) => (value, key) => or(
  compareToSearch(search)(key),
  compareToSearch(search)(value)
)

export const filterTokens = (tokens, search) => {
  if (!search) return tokens

  return pickBy(searchRules(search), tokens)
}
