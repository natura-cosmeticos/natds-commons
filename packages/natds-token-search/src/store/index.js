export * from './state'
export * from './tokens'
export * from './actions'
