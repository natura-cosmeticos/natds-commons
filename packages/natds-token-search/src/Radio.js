import { createElement } from './helpers'

export const Radio = (name, value, checked) => {
  const label = createElement('label', { for: value }, value)

  const input = createElement('input', {
    id: value,
    name,
    type: 'radio',
    value,
    ...(checked ? { checked: 'checked' } : {})
  })

  const wrapper = createElement('span')

  wrapper.appendChild(input)
  wrapper.appendChild(label)

  return wrapper
}

export default Radio
