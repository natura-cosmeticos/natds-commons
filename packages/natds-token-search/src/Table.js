import {
  pipe, join, mapObjIndexed, values
} from 'ramda'
import { createElement } from './helpers'
import { store } from './store'
import jss from './styles/jss'
import { color } from './styles/global'

const tableStyles = {
  tokenTable: {
    '& .deprecated': {
      backgroundColor: 'red',
      borderRadius: 8,
      color: 'white',
      fontSize: 12,
      padding: [0, 8]
    },
    '& .opacity': {
      backgroundColor: '#D51BE5',
      display: 'inline-block',
      height: '100%',
      width: '50%'

    },
    '& .opacity-wrapper': {
      backgroundColor: '#eaeaea',
      textAlign: 'center'
    },
    '& .preview': {
      border: '1px solid #eaeaea',
      height: 24,
      width: '100%'
    },
    '& td, & th': {
      border: [1, 'solid', color.gray],
      fontSize: 16,
      lineHeight: 1.5,
      padding: 8
    },
    '& tr': {
      textAlign: 'left'
    },
    borderCollapse: 'collapse',
    borderSpacing: 0,
    marginTop: 50,
    tableLayout: 'fixed',
    width: '100%'
  }
}

const tableSheet = jss.createStyleSheet(tableStyles)

const tableHeading = '<thead><th>Name</th><th>Value</th><th>Preview</th></thead>'

const buildNamePreview = (name) => {
  const { deprecatedTokens } = store.getState()

  if (Object.keys(deprecatedTokens).includes(name)) {
    return `<td>${name} <span class="deprecated">deprecated</span></td>`
  }

  return `<td>${name}</td>`
}

const buildPreview = (name, value) => {
  if (name.includes('color')) {
    return `<div class="preview color" style="background-color: ${value.replace(/"/g, '')}"></div>`
  }

  if (name.includes('opacity')) {
    return `<div class="preview opacity-wrapper"><span class="opacity" style="opacity: ${value}"></span></div>`
  }

  return ''
}

const renderBody = pipe(
  mapObjIndexed((value, name) => `<tr>${buildNamePreview(name)}<td>${value}</td><td>${buildPreview(name, value)}</td></tr>`),
  values,
  join('')
)

export const Table = () => {
  tableSheet.attach()
  const table = createElement('table', null, tableHeading, tableSheet.classes.tokenTable)
  const tableBody = createElement('tbody')

  store.subscribe(() => {
    tableBody.innerHTML = renderBody(store.getState().selectedTokens)
  })
  tableBody.innerHTML = renderBody(store.getState().selectedTokens)
  table.appendChild(tableBody)

  return table
}

export default Table
