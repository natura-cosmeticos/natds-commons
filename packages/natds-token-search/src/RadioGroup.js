import { createElement } from './helpers'
import jss from './styles/jss'
import { color } from './styles/global'
import { Radio } from './Radio'

const radioGroupStyles = {
  radioGroup: {
    '&.brand,&.mode': {
      '& input:checked+label': {
        borderColor: color.primary,
        color: 'black',
        fontWeight: 'bold'
      },
      '& label': {
        '&:hover': {
          backgroundColor: color.primaryHover,
          borderColor: color.primary
        },
        border: [1, 'solid', color.gray],
        borderRadius: 25,
        color: color.text,
        padding: [8, 24]
      }
    },
    '&.platform': {
      '& $radios': {
        '&::after': {
          borderBottom: [1, 'solid', color.gray],
          bottom: 0,
          content: '""',
          left: 16,
          position: 'absolute',
          right: 0
        },
        marginLeft: -20,
        position: 'relative'
      },
      '& h5': {
        display: 'none'
      },
      '& input': {
        '&:checked+label': {
          fontWeight: 'bold'
        },
        '&:checked+label::after': {
          borderBottom: [2, 'solid', color.primary],
          bottom: 0,
          content: '""',
          left: 0,
          position: 'absolute',
          right: 0
        }
      },
      '& label': {
        padding: [20, 0]
      },
      '& span': {
        display: 'flex',
        margin: [0, 16],
        padding: 0,
        width: 'auto'
      }
    },
    margin: [20, 0]
  },
  radios: {
    '& span': {
      '& *': {
        verticalAlign: 'middle'
      },
      '& input': {
        margin: 0,
        opacity: 0,
        position: 'absolute'
      },
      '& label': {
        fontSize: 14,
        position: 'relative'
      },
      marginRight: 24,
      padding: [8, 0]
    },
    display: 'flex'
  }

}

const radioGroupSheet = jss.createStyleSheet(radioGroupStyles)

export const RadioGroup = (name, data, selected, titleText) => {
  radioGroupSheet.attach()
  const wrapper = createElement('div', null, null, radioGroupSheet.classes.radioGroup)
  const title = createElement('h5', null, titleText)

  wrapper.classList.add(name)
  wrapper.appendChild(title)

  const radios = createElement('div', null, null, radioGroupSheet.classes.radios)

  data.forEach((item) => {
    radios.appendChild(Radio(name, item, selected === item))
  })

  wrapper.appendChild(radios)

  return wrapper
}

export default RadioGroup
