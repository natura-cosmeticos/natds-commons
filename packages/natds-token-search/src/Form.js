import { createElement } from './helpers'
import { searchAction, store } from './store'
import { RadioGroup } from './RadioGroup'
import { Search } from './Search'

const brands = [
  'aesop',
  'avon',
  'biome',
  'natura',
  'theBodyShop'
]

const modes = ['light', 'dark']

const platforms = [
  'android',
  'ios',
  'reactNative',
  'web'
]

export const onInputSearch = ({ target }) => store.dispatch(
  searchAction({ search: target.value })
)

const onFormChange = (event) => store.dispatch(
  searchAction({ [event.target.name]: event.target.value })
)

export const onFormSubmit = (event) => event.preventDefault()

export const Form = () => {
  const form = createElement('form')

  const state = store.getState()

  form.appendChild(RadioGroup('platform', platforms, state.platform, 'Platforms'))
  form.appendChild(Search(onInputSearch))
  form.appendChild(RadioGroup('brand', brands, state.brand, 'Brands'))
  form.appendChild(RadioGroup('mode', modes, state.mode, 'Modes'))

  form.addEventListener('change', onFormChange)
  form.addEventListener('submit', onFormSubmit)

  return form
}

export default Form
