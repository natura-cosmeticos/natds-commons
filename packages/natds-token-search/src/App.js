import { Table } from './Table'
import { Form } from './Form'
import { createElement } from './helpers'
import jss from './styles/jss'

const containerStyles = {
  container: {
    padding: 20
  }
}

const sheet = jss.createStyleSheet(containerStyles)

export const App = () => {
  sheet.attach()
  const wrapper = createElement('div', null, null, sheet.classes.container)

  wrapper.appendChild(Form())
  wrapper.appendChild(Table())

  return wrapper
}

export default App
