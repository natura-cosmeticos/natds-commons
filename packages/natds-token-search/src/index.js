import { App } from './App'
import { globalStyles, reset } from './styles/global'
import jss from './styles/jss'

jss
  .createStyleSheet(reset)
  .attach()

jss
  .createStyleSheet(globalStyles)
  .attach()

if (module.hot) module.hot.accept()

const root = document.getElementById('app-root')

root.appendChild(App())
