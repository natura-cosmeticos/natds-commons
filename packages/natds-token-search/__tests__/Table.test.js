import { store } from '../src/store'
import { Table } from '../src/Table'

const allTokens = {
  borderRadius: 8,
  color: '#ffffff',
  opacity: 1
}

const deprecatedTokens = {
  borderRadius: 8
}

const state = {
  allTokens,
  brand: 'natura',
  deprecatedTokens,
  mode: 'light',
  platform: 'web',
  search: null,
  selectedTokens: allTokens
}

describe('Table', () => {
  beforeEach(() => {
    jest.spyOn(store, 'getState').mockReturnValue(state)
  })
  it('should render table', () => {
    const table = Table()

    expect(table).toMatchSnapshot()
  })

  it('should update table on store changes', () => {
    const table = Table()

    jest.spyOn(store, 'getState').mockReturnValue({ ...state, selectedTokens: { color: '#ffffff' } })
    store.dispatch({ type: 'ACTION' })

    expect(table).toMatchSnapshot()
  })
})
