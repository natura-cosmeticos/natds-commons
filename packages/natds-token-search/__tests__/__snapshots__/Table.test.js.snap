// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render table 1`] = `
<table
  class="tokenTable-0-0-1"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Value
      </th>
      <th>
        Preview
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        borderRadius 
        <span
          class="deprecated"
        >
          deprecated
        </span>
      </td>
      <td>
        8
      </td>
      <td />
    </tr>
    <tr>
      <td>
        color
      </td>
      <td>
        #ffffff
      </td>
      <td>
        <div
          class="preview color"
          style="background-color: #ffffff"
        />
      </td>
    </tr>
    <tr>
      <td>
        opacity
      </td>
      <td>
        1
      </td>
      <td>
        <div
          class="preview opacity-wrapper"
        >
          <span
            class="opacity"
            style="opacity: 1"
          />
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table should update table on store changes 1`] = `
<table
  class="tokenTable-0-0-1"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Value
      </th>
      <th>
        Preview
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        color
      </td>
      <td>
        #ffffff
      </td>
      <td>
        <div
          class="preview color"
          style="background-color: #ffffff"
        />
      </td>
    </tr>
  </tbody>
</table>
`;
